src/lambda_function/s3_to_glue/app.py

```diff
@@ -86,20 +50,19 @@ def lambda_handler(event, context):
     Returns:
         (None) Submits new file records to a Glue workflow.
     """
-    s3 = boto3.client("s3")
-    source_bucket_name = os.environ["S3_SOURCE_BUCKET_NAME"]
-    trigger_event_date = parser.isoparse(event["time"]).date()
-    objects = s3.list_objects_v2(Bucket=source_bucket_name)
-    files_to_submit = []
-    if "Contents" in objects:
-        files_to_submit = query_files_to_submit(
-            objects, files_to_submit, trigger_event_date
+    messages = []
+    for record in event["Records"]:
+        bucket_name = record["s3"]["bucket"]["name"]
+        object_key = record["s3"]["object"]["key"]
```

Comment:
Just a note here: https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-content-structure.html.  It was helpful to know the S3 event notification structure.

One thing I'm not fully understanding is that this seems to just take in all the new records within a bucket without specific folders.  Based on what you were saying, am I understanding correctly that we would set up S3 notifications specific to different folders?

Reply:
The trigger can be configured to only alert for objects under a specific prefix, e.g., `{namespace}/raw_data/...`